src/net_processing.cpp

```diff
@@ -798,6 +803,11 @@ void PeerLogicValidation::BlockChecked(const CBlock& block, const CValidationSta
                 Misbehaving(it->second.first, nDoS);
         }
     }
+    else if (state.IsValid() && !IsInitialBlockDownload()) {
+        if (it != mapBlockSource.end() && State(it->second.first)) {
+            MaybeSetPeerAsAnnouncingHeaderAndIDs(it->second.first, *connman);
```

Comment:
Could we change the lifetime of mapBlockSource to keep it around until then? We could sweep it later.